src/test/app/Check_test.cpp

```diff
@@ -1848,22 +1883,741 @@ class Check_test : public beast::unit_test::suite
         env.require(balance(bob, drops(1'299'999'940)));
     }
 
+    void
+    testTrustLineCreation(FeatureBitset features)
+    {
+        // Explore automatic trust line creation when a check is cashed.
+        //
+        // This capability is enabled by the featureCheckCashMakesTrustLine
+        // amendment.  So most of this test executes only when that amendment
+        // is active.
+        testcase("Trust Line Creation");
+
+        using namespace test::jtx;
+
+        Env env{*this, features};
+
+        // An account that independently tracks its owner count.
+        struct AccountOwns
+        {
+            beast::unit_test::suite& suite;
+            Env const& env;
+            Account const acct;
+            std::size_t owners;
+
+            void
+            verifyOwners(std::uint32_t line) const
+            {
+                suite.expect(
+                    ownerCount(env, acct) == owners,
+                    "Owner count mismatch",
+                    __FILE__,
+                    line);
+            }
+
+            // Operators to make using the class more convenient.
+            operator Account const() const
+            {
+                return acct;
+            }
+
+            operator ripple::AccountID() const
+            {
+                return acct.id();
+            }
+
+            IOU
+            operator[](std::string const& s) const
+            {
+                return acct[s];
+            }
+        };
+
+        AccountOwns alice{*this, env, "alice", 0};
+        AccountOwns bob{*this, env, "bob", 0};
+
+        // Fund with noripple so the accounts do not have any flags set.
+        env.fund(XRP(5000), noripple(alice, bob));
+        env.close();
+
+        // First a sanity check.  If featureCheckCashMakesTrustLine is not
+        // live, then cashing a check without previously establishing a trust
+        // line fails.
+        {
+            AccountOwns gw1{*this, env, "gw1", 0};
+
+            // Fund gw1 with noripple (even though that's atypical for a
+            // gateway) so it does not have any flags set.  We'll set flags
+            // on gw1 later.
+            env.fund(XRP(5000), noripple(gw1));
+            env.close();
+
+            IOU const CK9 = gw1["CK9"];
+
+            Account const zoe{"zoe"};
+            env.fund(XRP(5000), zoe);
+            env.close();
+
+            uint256 const chkId{getCheckIndex(gw1, env.seq(gw1))};
+            env(check::create(gw1, zoe, CK9(100)));
+            env.close();
+
+            env(check::cash(zoe, chkId, CK9(100)),
+                ter(features[featureCheckCashMakesTrustLine]
+                        ? TER(tesSUCCESS)
+                        : TER(tecNO_LINE)));
+            env.close();
+        }
+
+        // All remaining tests require featureCheckCashMakesTrustLine.
+        if (!features[featureCheckCashMakesTrustLine])
+            return;
```

Comment:
With CheckCashMakesTrustLine disabled, this function is only the test above this return, which duplicates a test elsewhere. I personally find it easier to understand if this function is only run with CheckCashMakesTrustLine enabled, without lines 1943 - 1974 (the sanity check and early return), but maybe with an assertion that the feature is enabled.